spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java

```diff
@@ -0,0 +1,821 @@
+/*
+ * Copyright 2012-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.kafka;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.core.io.Resource;
+import org.springframework.kafka.listener.AbstractMessageListenerContainer.AckMode;
+
+/**
+ * Spring for Apache Kafka Auto-configuration properties.
+ *
+ * Users should refer to kafka documentation for complete descriptions of these
+ * properties.
+ *
+ * @author Gary Russell
+ * @since 1.5
+ */
+@ConfigurationProperties(prefix = "spring.kafka")
+public class KafkaProperties {
+
+	private final Consumer consumer = new Consumer();
+
+	private final Producer producer = new Producer();
+
+	private final Listener listener = new Listener();
+
+	private final Template template = new Template();
+
+	// Apache Kafka Common Properties
+
+	/**
+	 * A comma-delimited list of host:port pairs to use for establishing the initial
+	 * connection to the Kafka cluster.
+	 */
+	private List<String> bootstrapServers = Collections.singletonList("localhost:9092");
+
+	/**
+	 * An id to pass to the server when making requests; used for server-side logging.
+	 */
+	private String clientId;
+
+	/**
+	 * The password of the private key in the key store file.
```

Comment:
Meta-data description never starts with `The`, `A`, etc. Could you please align the documentation of those keys so that they match what we already have?